packages/core-ethereum/crates/core-ethereum-rpc/src/rpc.rs

```diff
@@ -0,0 +1,381 @@
+use async_trait::async_trait;
+use core_crypto::keypairs::{ChainKeypair, Keypair};
+use core_crypto::types::Hash;
+use core_ethereum_types::{ContractAddresses, ContractInstances};
+use ethers::middleware::{MiddlewareBuilder, NonceManagerMiddleware, SignerMiddleware};
+use ethers::prelude::k256::ecdsa::SigningKey;
+use ethers::prelude::transaction::eip2718::TypedTransaction;
+use ethers::signers::{LocalWallet, Signer, Wallet};
+use ethers::types::BlockId;
+use ethers_providers::{JsonRpcClient, Middleware, Provider, RetryClient, RetryClientBuilder, RetryPolicy};
+use primitive_types::H160;
+use serde::{Deserialize, Serialize};
+use std::sync::Arc;
+use std::time::Duration;
+use utils_types::primitives::{Address, Balance, BalanceType, U256};
+use validator::Validate;
+
+use crate::errors::Result;
+use crate::HoprRpcOperations;
+
+#[cfg(feature = "prometheus")]
+use utils_metrics::metrics::SimpleCounter;
+
+#[cfg(feature = "prometheus")]
+lazy_static::lazy_static! {
+     pub(crate) static ref METRIC_COUNT_RPC_CALLS: SimpleCounter = SimpleCounter::new(
+        "core_ethereum_counter_rpc_calls",
+        "Number of RPC calls"
+    )
+    .unwrap();
+}
+
+/// Configuration of the RPC related parameters.
+#[derive(Clone, Debug, Copy, PartialEq, Eq, Serialize, Deserialize, Validate)]
+pub struct RpcOperationsConfig {
+    /// Blockchain id
+    /// Default is 100.
+    pub chain_id: u64,
+    /// Addresses of all deployed contracts
+    /// Default contains empty (null) addresses.
+    pub contract_addrs: ContractAddresses,
+    /// Number of HTTP retries on retry-able failures
+    /// Defaults to 5
+    pub max_http_retries: u32,
+    /// Expected block time of the blockchain
+    /// Defaults to 5 seconds
+    pub expected_block_time: Duration,
+    /// Single log fetch chunk size
+    /// Defaults to 50
+    pub logs_page_size: u64,
+    /// Interval for polling on TX submission
+    /// Defaults to 7 seconds.
```

Comment:
Do you think these values will not get out of sync with the code easily? Would it maybe be more reasonable to write a docstring for the default implementation itself?

Reply:
We will tie the config defaults to the YAML config defaults, once we introduce this config structure in the file too.